Sources/BraintreeThreeDSecure/BTPaymentFlowDriver+ThreeDSecure.m

```diff
@@ -52,6 +52,10 @@ - (void)performThreeDSecureLookup:(BTThreeDSecureRequest *)request
             requestParameters[@"dfReferenceId"] = request.dfReferenceId;
         }
 
+        if (request.accountTypeAsString) {
```

Comment:
Do we need the `accountTypeAsString` property? Is it easier to read having the logic be here like this:   

```
if (request.accountType) {
    case BTThreeDSecureAccountTypeCredit:
         requestParameters[@"accountType"] = @"credit";
    case BTThreeDSecureAccountTypeDebit:
         requestParameters[@"accountType"] = @"debit";
    default:
         requestParameters[@"accountType"] = nil;
}
```

Reply:
Yeah, I can see why that might seem simpler. Here's my rationale for the `accountTypeAsString` property.

1) It makes the testing story simpler. If we test `accountTypeAsString` independently, we don't need to add extra tests for `performThreeDSecureLookup` for each path through the switch statement.

2) This type of if-statement:

```objc
if (request.parameter) {
    requestParameters[@"param"] = request.parameter;
}
```

is the pattern we follow with all of the other properties, so throwing a switch statement in there might actually make things less readable.

3) This particular enum only has three cases, but if we add an enum for `shippingMethod`, it'll have a lot of cases. We'll want to handle both enums in the same way, and having two switch statements in the `performThreeDSecureLookup` method doesn't seem super readable to me. I think having `asString` methods for both is the better way to go.